### PR TITLE
Change the README for a better Getting Start

### PR DESCRIPTION
**Pull request by ChelesteWang**
I tried to adjust the structure of the original readme so that beginners can better try the project

### CODE REVIEW

**Comment by kocoler** [comment]
This makes the README more detailed!
LGTM

**Review by kocoler — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,41 +11,77 @@ https://forum.casbin.com/
...
-- Get the code:
+## Getting Started
 
-    ```shell
+### Run back-end applications
+
+> Before continuing, please make golang development environment are properly installed on your machine.
+
+- Get the code:
+    form github
```
Comment (uran0sH) [inline README.md:27]: `from github`, not `form`

File: README.md [hunk 2]
```diff
@@ -61,79 +97,79 @@ Backend | RESTful API backend for Casbin-forum | Golang + Beego + MySQL | https:
         You could get them by clicking on this url: https://github.com/settings/developers
         You should set `Homepage URL` to fit your own domain address, for local testing, set`http://localhost:3000`. And set the `Authorization callback URL`, the same domain address as before, add `/callback/github` after that, for local testing, set`http://localhost:3000/callback/github`.
 
-    And to improve security, you could set a `state` value determined by **yourself** to make sure the request is requesting by yourself, such as "random".
-    Those information strings can be specified at: https://github.com/casbin/casbin-forum/blob/master/conf/app.conf
+        And to improve security, you could set a `state` value determined by **yourself** to make sure the request is requesting by yourself, such as "random".
```
Comment (uran0sH) [inline README.md:100]: maybe `requested by yourself`?

File: README.md [hunk 3]
```diff
@@ -61,79 +97,79 @@ Backend | RESTful API backend for Casbin-forum | Golang + Beego + MySQL | https:
...
+        GoogleAuthClientID = "xxx" //your own client id
+        GoogleAuthClientSecret = "xxx" //your own client secret
+        GoogleAuthState = "xxx" //set by yourself
+        GithubAuthClientID = "xxx" //your own client id
+        GithubAuthClientSecret = "xxx" //your own client secret
+        GithubAuthState = "xx" //set by yourself, we may change this to a random word in the future
+        ```
 
-    You may also have to fill in the **same** information at: https://github.com/casbin/casbin-forum/blob/master/web/src/Conf.js. By the way, you could change the value of `scope` to get different user information form them if you need, we just take `profile` and `email`.
+        You may also have to fill in the **same** information at: https://github.com/casbin/casbin-forum/blob/master/web/src/Conf.js. By the way, you could change the value of `scope` to get different user information form them if you need, we just take `profile` and `email`.
```
Comment (uran0sH) [inline README.md:112]: `from them`

**Review by uran0sH — CHANGES_REQUESTED** [review]

**Comment by ChelesteWang** [comment]
Thanks for helping correct the mistake

**Pull request closed without merging**